.github/workflow/multi-arch-build.yaml

```diff
@@ -0,0 +1,208 @@
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+        with:
+          driver-opts: network=host
+          install: true
+
+      - name: Build and locally push Skopeo
+        uses: docker/build-push-action@v2
+        with:
+          context: contrib/skopeoimage/${{ matrix.source }}
+          file: ./contrib/skopeoimage/${{ matrix.source }}/Dockerfile
+          platforms: ${{ env.PLATFORMS }}
+          push: true
+          tags: localhost:5000/skopeo/${{ matrix.source }}
+
+      # Simple verification that stable images work, and
+      # also grab version number use in forming the FQIN.
+      - name: amd64 container sniff test
+        if: matrix.source == 'stable'
+        id: sniff_test
+        run: |
+          VERSION_OUTPUT="$(docker run localhost:5000/skopeo/${{ matrix.source }} \
+                            skopeo --storage-driver=vfs version)"
+          echo "$VERSION_OUTPUT"
+          VERSION=$(grep -Em1 '^Version: ' <<<"$VERSION_OUTPUT" | awk '{print $2}')
+          test -n "$VERSION"
+          echo "::set-output name=version::${VERSION}"
+
+      - name: Generate skopeo reg. image FQIN(s)
+        id: skopeo_reg
+        run: |
+          if [[ "${{ matrix.source }}" == 'stable' ]]; then
+            # The `skopeo version` in image just built
+            VERSION='v${{ steps.sniff_test.outputs.version }}'
+            # workaround vim syntax-highlight bug: '
+            # Image tags previously pushed to quay
+            ALLTAGS=$(skopeo list-tags \
+                      docker://$SKOPEO_QUAY_REGISTRY/stable | \
+                      jq -r '.Tags[]')
+
+            # New image? Push quay.io/skopeo/stable:vX.X.X and :latest
+            if ! fgrep -qx "$VERSION" <<<"$ALLTAGS"; then
+              # Assume version-tag is also the most up to date (i.e. "latest")
+              FQIN="$SKOPEO_QUAY_REGISTRY/stable:$VERSION,$SKOPEO_QUAY_REGISTRY/stable:latest"
+            else # Not a new version-tagged image
+              # Assume other contents changed, so this is the "new" latest.
+              FQIN="$SKOPEO_QUAY_REGISTRY/stable:latest"
+            fi
+          elif [[ "${{ matrix.source }}" == 'testing' ]]; then
+            # Assume some contents changed, always push latest testing.
+            FQIN="$SKOPEO_QUAY_REGISTRY/testing:latest"
+          elif [[ "${{ matrix.source }}" == 'upstream' ]]; then
+            # Assume some contents changed, always push latest upstream.
+            FQIN="$SKOPEO_QUAY_REGISTRY/upstream:latest"
+          else
+            echo "::error::Unknown matrix item '${{ matrix.source }}'"
+            exit 1
+          fi
+          echo "::warning::Pushing $FQIN"
+          echo "::set-output name=fqin::${FQIN}"
+          echo '::set-output name=push::true'
+
+      # This is substantially the same as the above step, except the
+      # $CONTAINERS_QUAY_REGISTRY is used and the "testing"
+      # flavor is never pushed.
```

Comment:
Why isn't the testing flavor pushed?  A `testing:latest` should be pushed everytime this script runs me thinks.

Reply:
One is for `quay.io/skopeo`, just not `quay.io/containers` because "this is how it was before" :grin: I can change this behavior but...remember this config. is a copy-paste of what we already do in podman and buildah.  Should I change those also?

Reply:
Clarification: `quay.io/containers/skopeo` has tags for `latest` and `$VERSION` only.  This is different from `quay.io/skopeo` where we have an **image** called `testing`. So if we want to push testing to containers, I would need to add something like a `:testing` tag.

Reply:
sorry for the confusion.  I do NOT want a `:testing` label for any of our images.   I wasn't seeing that the `quay.io/skopeo/testing:latest` image was getting pushed as part of this processing.  Did I miss that?

Reply:
found it, this is good, ty!